### ISSUE TITLE
Access violation on right mous click when multiple files are selected

### ISSUE THREAD

**Issue by cbreemer** [doublecmd/doublecmd#1476]
### Version

Double Commander
Version: 1.0.11
Revision: 272
Commit: 0b8ba3838
Build date: 2023/04/02
Lazarus: 2.2.7.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 11 22H2 x86_64

### Description

When I select more than one file, then right mouse click, an access violation occurs. Entry in doublecmd.err :

--------------- 08-02-2024, 16:06:31 ---------------
| DC v1.0.11 Rev. 272 -- x86_64-Win64-win32/win64
| Windows 11 22H2 x86_64 | PID 22708
Unhandled exception: EAccessViolation: Access violation
  Stack trace:
  $00007FFFA40A774B in C:\Windows\System32\shell32.dll
  $00007FFFA40AA099 in C:\Windows\System32\shell32.dll
  $00007FFFA40AEDA7 in C:\Windows\System32\shell32.dll
  $00007FF7EDA88B81 line 557, column 95 of platform/win/ushellcontextmenu.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED986DAC line 704, column 41 of platform/uosforms.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED96FEED line 784, column 9 of umaincommands.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7EDA1AD0A line 313, column 11 of fileviews/ufileviewwithgrid.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED80A54F line 2347, column 5 of include/control.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED80B2EF line 2850, column 20 of include/control.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED680811 in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED80A351 line 2287, column 24 of include/control.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED7FE609 line 5433, column 28 of include/wincontrol.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED8D047D line 4949, column 31 of grids.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED8C69A9 line 112, column 36 of lclmessageglue.pas in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED7E0DC9 line 2609, column 38 of win32/win32callback.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FF7ED7E1560 line 2774, column 19 of win32/win32callback.inc in C:\Program Files\Double Commander\doublecmd.exe
  $00007FFFA4BB8241 in C:\Windows\System32\user32.dll

Interestingly, I have the same crash when I perform the exact same action in the Windows Explorer : 

Faulting application name: explorer.exe, version: 10.0.22621.3085, time stamp: 0xa58eee1c
Faulting module name: SHELL32.dll, version: 10.0.22621.3085, time stamp: 0x4e9c34b7
Exception code: 0xc0000005
Fault offset: 0x000000000008774b
Faulting process id: 0x0x1BE8
Faulting application start time: 0x0x1DA569D0B6275D7
Faulting application path: C:\Windows\explorer.exe
Faulting module path: C:\Windows\System32\SHELL32.dll
Report Id: 0ea2bebf-9e88-4d3c-a25b-431d2a231063
Faulting package full name: 
Faulting package-relative application ID: 

So it is perhaps not a double commander issue.

**Comment by JonnyTech** [comment]
It is caused by a malfunctioning shell extension. You can use [ShellExView](https://www.nirsoft.net/utils/shexview.html) to identify and remove which one.

**Comment by cbreemer** [comment]
Thanks for your reply ! I had already found this tip elsewhere, but even disabling all (non-Microsoft) extensions did not help.
I thought of removing them but the options under the "Remove" menu item are all greyed out even though I'm running ShellExView as Administrator.
So I uninstalled nearly all software responsible for these extensions (Setpoint, 7-Zip, Foxit, Avast) but the issue remains. There's only two non-MS extensions
listed now, one from Foxit (which is no longer there) and one from Edge (which I cannot uninstall apparently).

I am scared to start messing around with the MS extensions for fear of buggering up the system. Hmm....

But this is clearly not a DoubleCommander issue so I'm happy for you to close the ticket (unless you want to persevere 😁).

Cheers,
Chris Breemer



________________________________
